packages/astro/src/types/public/config.ts

```diff
@@ -95,6 +95,26 @@ export type ServerConfig = {
 	 * ```
 	 */
 	open?: string | boolean;
+
+	/**
+	 * @name server.allowedHosts
+	 * @type {string[] | true}
+	 * @default `false`
+	 * @version 5.4.0
+	 * @description
+	 * 
+	 * A list of hostnames that Astro is allowed to respond to. When the value is set to `true`, any 
+	 * hostname is allowed. 
+	 *
+	 * ```js
+	 * {
+	 *   server: { 
+	 *   	allowedHosts: ['staging.example.com', 'qa.example.com']
+	 *   }
+	 * }
+	 * ```
+	 */
+	allowedHosts?: string[] | boolean;
```

Comment:
Shouldn't we use `true` here instead of `boolean`? Since the type is `z.literal(true)`, when using `false`, Typescript will throw an error, right?
```suggestion
	allowedHosts?: string[] | true;
```
Also, it seems we use `false` instead of `boolean` for `markdown.syntaxHighlight` which uses `z.literal(false)`. So it would be consistent.